discretisedfield/field.py

```diff
@@ -3435,152 +3435,224 @@ def _hv_key_dims(self):
+            ft_vdims = [f"ft_{vdim}" for vdim in self.vdims]
 
-        values = []
-        for idx in range(self.nvdim):
-            ft = np.fft.fftshift(np.fft.fftn(self.array[..., idx].squeeze()))
-            values.append(ft.reshape(mesh.n))
+            ft_vdim_mapping = {}
+            for vdim, ft_vdim in zip(self.vdims, ft_vdims):
+                if vdim in self.vdim_mapping.keys():
+                    ft_vdim_mapping[ft_vdim] = "k_" + self.vdim_mapping[vdim]
+        else:
+            ft_vdims = None
+            ft_vdim_mapping = None
 
         return self.__class__(
             mesh,
-            nvdim=len(values),
-            value=np.stack(values, axis=3),
-            vdims=self.vdims,
-            # TODO vdim_mapping with fft_mesh dims
+            nvdim=self.nvdim,
+            value=ft,
+            vdims=ft_vdims,
+            unit=self.unit,
+            vdim_mapping=ft_vdim_mapping,
         )
 
-    @property
     def ifftn(self):
         """Inverse Fourier transform.
 
+        N dimentional discrete inverse FFT of the field.
+
         Returns
         -------
         discretisedfield.Field
+
+            Inverse Fourier transformed field.
+
+        Examples
+        --------
+        1. Create a mesh and perform an iFFT.
+        >>> import discretisedfield as df
+        >>> mesh = df.Mesh(p1=0, p2=10, cell=2)
+        >>> field = df.Field(mesh, dim=3, value=(1, 2, 3))
+        >>> ifft_field = field.fftn().ifftn()
+        >>> ifft_field.nvdim
+        array([3])
+        >>> ifft_field.vdims
+        ('x', 'y', 'z')
+
         """
-        mesh = self.mesh.attributes["realspace_mesh"]
-        if self.mesh.attributes["isplane"] and not mesh.attributes["isplane"]:
-            mesh = mesh.plane(self.mesh.region.dims[self.mesh.attributes["planeaxis"]])
+        mesh = self.mesh.ifftn()
+
+        axes = np.arange(self.mesh.region.ndim)
+        ft = np.fft.ifftn(
+            np.fft.ifftshift(self.array, axes=axes),
+            axes=axes,
+        )
 
-        values = []
-        for idx in range(self.nvdim):
-            ft = np.fft.ifftn(np.fft.ifftshift(self.array[..., idx].squeeze()))
-            values.append(ft.reshape(mesh.n))
+        if self.vdims is not None:
+            vdims = [ftvdim[3:] for ftvdim in self.vdims]
+
+            vdim_mapping = {}
+            for vdim, ft_vdim in zip(vdims, self.vdims):
+                if ft_vdim in self.vdim_mapping.keys():
+                    vdim_mapping[vdim] = self.vdim_mapping[ft_vdim][2:]
+        else:
+            vdims = None
+            vdim_mapping = None
 
         return self.__class__(
             mesh,
-            nvdim=len(values),
-            value=np.stack(values, axis=3),
-            vdims=self.vdims,
-            # TODO vdim_mapping
+            nvdim=self.nvdim,
+            value=ft,
+            vdims=vdims,
+            unit=self.unit,
+            vdim_mapping=vdim_mapping,
         )
 
-    @property
     def rfftn(self):
```

Comment:
Do we need `rfft` and `irfft` in `mag2exp` (or anywhere else in ubermag)? If not I would suggest that we consider removing it. I would assume that `[i]fftn` should be enough functionality and we would then also avoid the "problem" with the mesh shape in `irfftn`.